apm-lambda-extension/main_test.go

```diff
@@ -0,0 +1,343 @@
+	case InvokeStandard:
+		time.Sleep(time.Duration(event.ExecutionDuration) * time.Second)
+		req, _ := http.NewRequest("POST", "http://localhost:8200/intake/v2/events", bytes.NewBuffer([]byte(event.APMServerBehavior)))
+		res, _ := client.Do(req)
+		log.Println(res.StatusCode)
+	case InvokeStandardFlush:
+		time.Sleep(time.Duration(event.ExecutionDuration) * time.Second)
+		reqData, _ := http.NewRequest("POST", "http://localhost:8200/intake/v2/events?flushed=true", bytes.NewBuffer([]byte(event.APMServerBehavior)))
+		client.Do(reqData)
+	case InvokeWaitgroupsRace:
+		time.Sleep(time.Duration(event.ExecutionDuration) * time.Second)
+		reqData0, _ := http.NewRequest("POST", "http://localhost:8200/intake/v2/events", bytes.NewBuffer([]byte(event.APMServerBehavior)))
+		reqData1, _ := http.NewRequest("POST", "http://localhost:8200/intake/v2/events", bytes.NewBuffer([]byte(event.APMServerBehavior)))
+		go client.Do(reqData0)
+		go client.Do(reqData1)
+		time.Sleep(650 * time.Microsecond)
+	case InvokeMultipleTransactionsOverload:
+		wg := sync.WaitGroup{}
+		for i := 0; i < 200; i++ {
+			go func() {
+				wg.Add(1)
+				time.Sleep(time.Duration(event.ExecutionDuration) * time.Second)
+				reqData, _ := http.NewRequest("POST", "http://localhost:8200/intake/v2/events", bytes.NewBuffer([]byte(event.APMServerBehavior)))
+				client.Do(reqData)
+				wg.Done()
+			}()
+		}
+		wg.Wait()
+	case Shutdown:
+		reqData, _ := http.NewRequest("POST", "http://localhost:8200/intake/v2/events", bytes.NewBuffer([]byte(event.APMServerBehavior)))
+		client.Do(reqData)
+	}
+	sendLogEvent(currId, "platform.runtimeDone")
+}
+
+func sendNextEventInfo(id string, event MockEvent) []byte {
+	nextEventInfo := extension.NextEventResponse{
+		EventType:          "INVOKE",
+		DeadlineMs:         time.Now().UnixNano()/int64(time.Millisecond) + int64(event.Timeout*1000),
+		RequestID:          id,
+		InvokedFunctionArn: "arn:aws:lambda:eu-central-1:627286350134:function:main_unit_test",
+		Tracing:            extension.Tracing{},
+	}
+	if event.Type == Shutdown {
+		nextEventInfo.EventType = "SHUTDOWN"
+	}
+
+	out, _ := json.Marshal(nextEventInfo)
+	return out
+}
+
+func sendLogEvent(requestId string, logEventType string) {
+	record := logsapi.LogEventRecord{
+		RequestId: requestId,
+	}
+	logEvent := logsapi.LogEvent{
+		Time:   time.Now(),
+		Type:   logEventType,
+		Record: record,
+	}
+	logEvent.RawRecord, _ = json.Marshal(logEvent.Record)
+	body, _ := json.Marshal([]logsapi.LogEvent{logEvent})
+	req, _ := http.NewRequest("POST", "http://localhost:8205", bytes.NewBuffer(body))
+	client := http.Client{}
+	client.Do(req)
+}
+
+func eventQueueGenerator(inputQueue []MockEvent, eventsChannel chan MockEvent) {
+	for _, event := range inputQueue {
+		eventsChannel <- event
+	}
+}
+
+// TESTS
+
+func TestStandardEventsChain(t *testing.T) {
+	http.DefaultServeMux = new(http.ServeMux)
+	log.Println("Standard Test")
+
+	eventsChannel := make(chan MockEvent, 100)
+	defer close(eventsChannel)
```

Comment:
we probably don't need to close `eventsChannel` explicitly, unless there's a reason to?

Reply:
Indeed, solved in [d9a8edc](https://github.com/elastic/apm-aws-lambda/pull/136/commits/d9a8edc0ca29cd924a0217ed22f9715a5c270c90)